src/main/java/org/betonquest/betonquest/compatibility/protocollib/api/GlowAPI.java

```diff
@@ -0,0 +1,238 @@
+package org.betonquest.betonquest.compatibility.protocollib.api;
+
+import com.comphenix.protocol.PacketType;
+import com.comphenix.protocol.ProtocolLibrary;
+import com.comphenix.protocol.events.PacketContainer;
+import com.comphenix.protocol.utility.MinecraftReflection;
+import com.comphenix.protocol.wrappers.WrappedDataWatcher;
+import com.comphenix.protocol.wrappers.WrappedWatchableObject;
+import lombok.CustomLog;
+import net.citizensnpcs.api.CitizensAPI;
+import net.citizensnpcs.api.npc.NPC;
+import org.bukkit.ChatColor;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Glow API class
+ */
+@CustomLog
+public class GlowAPI {
+
+    /**
+     * Constructor of Glow API instance
+     */
+    public GlowAPI() {
+    }
+
+    /**
+     * Creating and Sending Glow Packet for NPC in async (background).
+     *
+     * @param npcId   the id of NPC
+     * @param color   Set color for glowing entity
+     * @param glowing true if entity need to be glowing
+     * @param players List of players that can see the glowing NPC
+     * @return void
+     */
+    public CompletableFuture<Void> glowPacketAsync(final int npcId, final ChatColor color, final boolean glowing, final Collection<? extends Player> players) {
+        final NPC npc = CitizensAPI.getNPCRegistry().getById(npcId);
+        if (npc == null) {
+            LOG.warn("NPC Glow could not update Glowing for npc " + npcId + ": No npc with this id found!");
+            return null;
+        }
+        final Entity entity = npc.getEntity();
```

Comment:
Two issues here;
1. this code is a duplicate of the code in line 64-70 -> create a utility method for this
2. This is code containing Citizens Code. I would strongly recommend to extract this code to the Citizens code (probably the NPCGlow class) and only pass the entity to this class